runtime/lua/vim/lsp/util.lua

```diff
@@ -1389,14 +1429,12 @@ function M.locations_to_items(locations)
   for _, uri in ipairs(keys) do
     local rows = grouped[uri]
     table.sort(rows, position_sort)
-    local bufnr = vim.uri_to_bufnr(uri)
-    vim.fn.bufload(bufnr)
     local filename = vim.uri_to_fname(uri)
     for _, temp in ipairs(rows) do
       local pos = temp.start
       local row = pos.line
-      local line = (api.nvim_buf_get_lines(bufnr, row, row + 1, false) or {""})[1]
-      local col = M.character_offset(bufnr, row, pos.character)
+      local line = M.get_line(uri, row)
```

Comment:
Micro opt, but would it be better to pre-sort the pairs by {uri1 = {line1, line2, etc}}, have a get_lines function, to avoid re-opening a file and iterating each time you get a single line? Not opposed to merging it as is, just putting it out there :)

Reply:
I actually did it like that first, but then changed it back for some reason I can't remember now. I'll add it again

Reply:
It's already plenty fast. If there's a good reason not to do it then I wouldn't worry about it :)

Reply:
The code is a lot simpler without retrieving multiple rows at the time. So yeah, maybe I should leave it as is for now :).

I'll just add in the `.` in that comment and will push my changes